src/sync/seg_queue.rs

```diff
@@ -89,6 +89,21 @@ impl<T> SegQueue<T> {
         }
     }
 
+    /// Judge if the queue is empty.
+    ///
+    /// Returns `true` if the queue is empty.
+    pub fn is_empty(&self) -> bool {
+        let guard = epoch::pin();
+        let head = self.head.load(Acquire, &guard).unwrap();
+        let tail = self.tail.load(Acquire, &guard).unwrap();
+        if head.as_raw() != tail.as_raw() {
+            return false;
+        }
+
+        let low = head.low.load(Relaxed);
```

Comment:
actually the logic just come from `try_pop()`, I didn't change the `Relaxed` logic here. But I think in line 97 we already has an **Acquire** sync which is enough.